app/components/record_details/view.rb

```diff
@@ -57,13 +57,18 @@ def region
     end
 
     def trn_row
-      return nil unless [trainee.trn, trainee.submitted_for_trn_at].any?(&:present?)
+      return nil unless [trainee.trn, trainee.submitted_for_trn_at, trainee.created_from_hesa].any?(&:present?)
 
       if trainee.trn.present?
         {
           field_label: t(".trn"),
           field_value: trainee.trn,
         }
+      elsif trainee.created_from_hesa
```

Comment:
cc @edwardhorsford I'm a bit confused about this one - TRN isn't an option field for 'trn_received' trainees.

Reply:
Not sure I follow what this is

Reply:
Ah apologies! If a HESA trainee doesn't have a TRN, it would now show the new "Not provided from HESA update" message.

Reply:
I worked out which fields are optional using this spreadsheet i made. it takes the definition of the fieldnames (the leftmost column and finds the ones where minlength < 0. https://docs.google.com/spreadsheets/d/1dFbo4eGbTwV8uJ9IXFiaJi6hpfpbpxGNBj8kFEI4QQI/edit#gid=0
Doesnt look like endate is an optional field here (admittedly that is pretty weird)

Reply:
i'm pretty sure we should be showing this message for itt_end_date (it's also the example screenshot in the Trello card). itt_end_date is missing for the vast majority of HESA trainees. maybe @AndrewPetrie1 would be a good person to chat to with regards to the data most commonly missing.

Reply:
Itt end date has now been added to the optional fields we display the message for